jscomp/others/js.ml

```diff
@@ -249,6 +249,9 @@ module Re = Js_re
 module Promise = Js_promise
 (** Provide bindings to JS Promise *)
 
+module Promise2 = Js_promise2
+(** Provide bindings to JS Promise *)
+
 module Date = Js_date
 (** Provide bindings for JS Date *)
 
```

Comment:
TODO: integrate orthogonal quality of life improvements from https://github.com/ryyppy/rescript-promise

TODO2: plan the whole Js.Foo2 story and how it should move to be the default instead. This cleanup could go into v11. For a separate issue.